### ISSUE TITLE
Uneven match error in Website Agent

### ISSUE THREAD

**Issue by HyShai** [cantino/huginn#990]
The uneven mathc error really causes issues when the data has 1 or more matches. E.g. trying to extract from this xml:

```xml
<item>
   <link>http://example.com</link>
  <enclosure url="http://example.com/foo.mp3" length="9032127" type="audio/mpeg"/>
  <enclosure url="http://example.com/bar.mp3" length="9032131" type="audio/mpeg"/>
</item>
```

using xpath 1.0 there doesn't seem to be a way to extract both `enclosure`s. 

Is it possible to instead do a check inside the loop for `nil`? e.g.

```ruby
#don't know if this is valid ruby....but - get the largest unique length
num_unique_lengths.max.times do |index|
  result = {}
  interpolated['extract'].keys.each do |name|
   
    #pseudo code - if that index doesn't exist then just set to empty string
    result[name] = output[name][index] || ""
    if name.to_s == 'url'
      result[name] = (response.env[:url] + result[name]).to_s
    end
  end
  
....

end

```

**Comment by cantino** [comment]
What selector are you trying to use @HyShai?

**Comment by HyShai** [comment]
@cantino doing something like this:

```json
  "extract": {
    "url": {
      "xpath": "//*/item/link",
      "value": ".//text()"
    },
    "mp3": {
      "xpath": "//*/item/enclosure",
      "value": "@url"
    }
  }
```

`mp3` returns multiple results but `url` only returns one result. In such a case, ideally I would like Huginn to not throw an error, instead it would create either a) multiple events or b) an array of values for `mp3`. This has been a common issue for me when using the Website Agent to extract information from a source that can have 1 to many relationships. E.g. 1 webpage (`url`) has multiple images (`img`).

**Comment by cantino** [comment]
@HyShai: I agree.  I think we should have a "group" option or something that could combine the multiple mp3 values into an array.

**Comment by knu** [comment]
I don't think we need a new feature.  If you want to create an event per enclosure, just do so:
```
  "url": {
      "xpath": "//*/item/enclosure",
      "value": "./parent::li/link/text()"
    },
```

**Comment by knu** [comment]
I didn't read the follow-ups carefully.  Yeah, currently there is no way to emit an array value for a key in an event.

**Comment by level20peon** [comment]
Try it this way: https://github.com/cantino/huginn/issues/1033#issuecomment-140639795

**Comment by HyShai** [comment]
@level20peon unfortunately that workaround doesn't work in a scenario where there is an arbitrary/unknown number of elements to be matched (the EventFormatting Agent doesn't use `String#scan` so it would [only return the first match](http://stackoverflow.com/a/7993732).)

I still think we need some way to allow the WebsiteAgent to extract uneven matches. This has been a real annoyance for me :(

**Comment by Toucan-Sam** [comment]
Has anyone been able to figure this out?

**Comment by dsander** [comment]
@Toucan-Sam If there is only one `item` using `array: true` will work. 
```
  "extract": {
    "url": {
      "xpath": "//*/item/link",
      "value": ".//text()"
    },
    "mp3": {
      "xpath": "//*/item/enclosure",
      "value": "@url",
      "array": "true",
    }
  }
```

**Comment by gingerbeardman** [comment]
@dsander array:true is very useful, but how to I deal with those arrays with fewer elements when they get to DataOutputAgent?

**Comment by dsander** [comment]
@gingerbeardman I guess it depends on the output format, when you are using liquid the [`as_object` filter ](https://github.com/huginn/huginn/wiki/Formatting-Events-using-Liquid#as_object) might help. In other cases I think you can use JSONpath to get the whole array.

**Comment by gingerbeardman** [comment]
Thanks! I'll look into that.
